### ISSUE TITLE
BUG : using square brackets instead of parentheses in compiled SQL

### ISSUE THREAD

**Issue by aleenprd** [Titan-Systems/titan#100]
`snowflake.connector.errors.ProgrammingError: 001003: failed to execute sql, [ALTER NETWORK POLICY TITAN_TEST_NETWORK_POLICY SET allowed_network_rule_list = ['TITAN_TEST_NETWORK_RULE']]`
=>
```
snowflake.connector.errors.ProgrammingError: 001003 (42000): SQL compilation error:
syntax error line 1 at position 79 unexpected '['.
```

Furthermore, you can't use direct resource object reference because it gets compiled to this SQL: 
`[ALTER NETWORK POLICY TITAN_TEST_NETWORK_POLICY SET allowed_network_rule_list = [Resource:TITAN_TEST_NETWORK_RULE]]`

**Comment by neehhaa06** [comment]
Hello!! @aleenprd :)

Replace the square brackets [] with parentheses () in the SQL statement to fix the syntax error:                                                               

```
Original:
ALTER NETWORK POLICY TITAN_TEST_NETWORK_POLICY SET allowed_network_rule_list = ['TITAN_TEST_NETWORK_RULE'];

Fixed: 
ALTER NETWORK POLICY TITAN_TEST_NETWORK_POLICY SET allowed_network_rule_list = ('TITAN_TEST_NETWORK_RULE');
```

**Comment by aleenprd** [comment]
@neehhaa06 

Uhm, it's how titan compiles the SQL, I am not writing the SQL myself. This is an error appearing on blueprint.apply. 

The resource is defined in my code as:
```
titan_test_network_policy = NetworkPolicy(
    name="TITAN_TEST_NETWORK_POLICY",
    comment="Demo network policy for testing purposes of titan[core].",
    owner="SYSADMIN",
    allowed_network_rule_list=[titan_test_network_rule],
    blocked_network_rule_list=None,
    allowed_ip_list=None,
    blocked_ip_list=None,
    database=titan_test_db,
    schema=titan_test_schema,
)
```

As per the titan [source code](https://github.com/Titan-Systems/titan/blob/cd7070991e4ba330adfc64e64e2a41ae0ebd4d26/titan/resources/network_policy.py#L49).

Am I misunderstanding something?

**Comment by neehhaa06** [comment]
>To fix the issue, ensure **titan_test_network_rule** is a string identifier in your code:

```
titan_test_network_rule = "TITAN_TEST_NETWORK_RULE"
```

> Update your NetworkPolicy definition accordingly. Check Titan’s documentation for correct resource reference formats. If problems persist, consult Titan support for further assistance.

**Comment by aleenprd** [comment]
@neehhaa06 I already tried that as well. The result is simply that it gets compiled to `snowflake.connector.errors.ProgrammingError: 001003: failed to execute sql, [ALTER NETWORK POLICY TITAN_TEST_NETWORK_POLICY SET allowed_network_rule_list = ['TITAN_TEST_NETWORK_RULE']]` like I have already stated in my original issue. The problem is that it compiles the array with brackets which is not SQL syntax. Again, this means it results in the following error: `snowflake.connector.errors.ProgrammingError: 001003: failed to execute sql, [ALTER NETWORK POLICY TITAN_TEST_NETWORK_POLICY SET allowed_network_rule_list = ['TITAN_TEST_NETWORK_RULE']]` Please read carefully. You can recreate using this snippet:

```
titan_test_network_rule = NetworkRule(
    name="TITAN_TEST_NETWORK_RULE",
    type="IPV4",
    value_list=["some_ipv4_address"],
    mode="INGRESS",
    comment="Demo network rule resticting access to home Wifi only.",
    owner="SYSADMIN",
    database=titan_test_db,
    schema=titan_test_schema,
)

titan_test_network_policy = NetworkPolicy(
    name="TITAN_TEST_NETWORK_POLICY",
    comment="Demo network policy for testing purposes of titan[core].",
    owner="SYSADMIN",
    allowed_network_rule_list=["TITAN_TEST_NETWORK_RULE"],
    blocked_network_rule_list=None,
    allowed_ip_list=None,
    blocked_ip_list=None,
    database=titan_test_db,
    schema=titan_test_schema,
)
```

Furthermore, the documentation literally states that you can use it like this: `allowed_network_rule_list=[NetworkRule(name="rule1"), NetworkRule(name="rule2")]` (see link I posted previously).
```
    def __init__(
        self,
        name: str,
        allowed_network_rule_list: list[NetworkRule] = None,
```
PS: this is the source code as this particular resource is not documented in your wiki anyway.

**Comment by teej** [comment]
Fixed in [v0.9](https://github.com/Titan-Systems/titan/releases/tag/v0.9.1)
